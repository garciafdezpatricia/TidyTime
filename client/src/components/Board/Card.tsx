import { useState } from "react";
import { TbArrowMoveRight } from "react-icons/tb";
import { uuid } from "uuidv4";
import { useTaskContext } from "../Context/TaskContext";
import MoveModal from "../Modal/AbsoluteModal/AbsoluteModal";

export interface CardProps {
    cardIndex: number; // card index
    title: string,
    list: string, // task list name
    columnIndex: number,
    handleMoveTask: (arg?:any, arg2?:any) => void | any;
}

export default function Card({cardIndex, title, list, columnIndex, handleMoveTask} : CardProps) {

    const {setBoardItems, boardColumns} = useTaskContext();

    return (
        <article key={cardIndex} className="board-column-content-item">
            <div className="item-header">
                <p>#{list}</p>
                <div className="move-to-icon">
                    <TbArrowMoveRight onClick={() => handleMoveTask(columnIndex, cardIndex)} size={"1.2rem"} className="move-to-icon-icon" />
                </div>
            </div>
            <p className="item-content">{title}</p>
        </article>
    )
}