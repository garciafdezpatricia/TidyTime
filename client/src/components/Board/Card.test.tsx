import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import Card from '@/src/components/Board/Card';

describe('Card', () => {
    it('renders without crashing', () => {
      const { container } = render(<Card cardIndex={0} title={""} list={""} taskId={""} columnIndex={0} handleMoveTask={function (arg?: any, arg2?: any) {
          throw new Error("Function not implemented.");
      } } handleCardClick={function (arg?: any) {
          throw new Error("Function not implemented.");
      } }/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Card cardIndex={0} title={""} list={""} taskId={""} columnIndex={0} handleMoveTask={function (arg?: any, arg2?: any) {
            throw new Error("Function not implemented.");
        } } handleCardClick={function (arg?: any) {
            throw new Error("Function not implemented.");
        } }/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });