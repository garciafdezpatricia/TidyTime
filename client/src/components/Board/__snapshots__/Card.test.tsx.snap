// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="board-column-content-item"
  >
    <div
      class="item-header"
    >
      <p>
        #
      </p>
      <div
        class="move-to-icon"
      >
        <svg
          class="move-to-icon-icon"
          cursor="pointer"
          fill="none"
          height="1.2rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1.2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Move task
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M11 12h10"
          />
          <path
            d="M18 9l3 3l-3 3"
          />
          <path
            d="M7 12a2 2 0 1 1 -4 0a2 2 0 0 1 4 0z"
          />
        </svg>
      </div>
    </div>
    <p
      class="item-content"
    />
  </article>
</DocumentFragment>
`;
